### DISCUSSION TITLE
Line Graph for Non-Aggregates

### DISCUSSION THREAD

**Discussion by danielmackay** [datalust/seq-tickets#1492] [Help] [answered]
Hi there,

I am trying to create a graph in a dashboard that plots on or more values as they change over time.  I have a handful of events logged and am trying to use a query similar to:

```
select MarketPrice
from stream 
where MarketPrice > 0
group by time(1m)
```

However, I when trying to use a line graph I get an error stating 'MarketPrice' is not an aggregate.  

I would like to plot the actual value, not an aggregate.  Is that possible?

Thanks in advance

**Comment by liammclennan** [comment, marked as answer]
It is not possible to plot the actual values with a grouping. 

You can do what you want on the events page:

```sql
select @Timestamp, MarketPrice
from stream 
where MarketPrice > 0
```

Then click the 'View as timeseries' button. 

Dashboards are always aggregated by time, so you have to supply an aggregate function. `mean(MarketPrice)` might be useful, or you can plot multiple values on a chart e.g. `min(MarketPrice), mean(MarketPrice), max(MarketPrice)`.

```sql
select min(MarketPrice), mean(MarketPrice), max(MarketPrice)
from stream 
where MarketPrice > 0
group by time(1m)
```

Hope that helps.

**Reply by danielmackay** [reply]
`@Timestamp` was the missing key.

Thanks so much for your help with this. 🙂

**Reply by liammclennan** [reply]
You're welcome Daniel. Glad I could help. 

`@Timestamp` is one of the 'built-in' properties available on all events. Here is the full list https://docs.datalust.co/docs/built-in-properties-and-functions.
